spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/LSTM.scala

```diff
@@ -0,0 +1,337 @@
+    val realWei_iter = MemoryData.operationWant2(fwdPD, Query.WeightsPd, 1)
+    val realBias = MemoryData.operationWant2(fwdPD, Query.WeightsPd, 2)
+
+    val realDst = MemoryData.operationWant2(fwdPD, Query.DstPd, 0)
+    val realDst_iter = MemoryData.operationWant2(fwdPD, Query.DstPd, 1)
+
+    require(src_i.size().product == realSrc_iter.shape.product,
+      s"${getName} src iter shape is not correct.")
+    require(dst_i.size().product == realDst_iter.shape.product,
+      s"${getName} dst iter shape is not correct.")
+    require(weight.size().product == realWei.shape.product,
+      s"${getName} weight shape is not correct.")
+    require(weight_i.size().product == realWei_iter.shape.product,
+      s"${getName} weight iter shape is not correct.")
+    require(bias.size().product == realBias.shape.product,
+      s"${getName} bias shape is not correct.")
+
+    weight.setMemoryData(HeapData(weight.size(), Memory.Format.ldigo), realWei, runtime)
+    weight_i.setMemoryData(HeapData(weight_i.size(), Memory.Format.ldigo), realWei_iter, runtime)
+    bias.setMemoryData(HeapData(bias.size(), Memory.Format.ldgo), realBias, runtime)
+    src_i.setMemoryData(HeapData(src_i.size(), Memory.Format.ldsnc), realSrc_iter, runtime)
+    dst_i.setMemoryData(HeapData(dst_i.size(), Memory.Format.ldsnc), realDst_iter, runtime)
+
+    weight.sync()
+    weight_i.sync()
+    bias.sync()
+    src_i.sync()
+    dst_i.sync()
+
+    val srcs = Array(realSrc.getPrimitive(runtime), realSrc_iter.getPrimitive(runtime),
+      realWei.getPrimitive(runtime), realWei_iter.getPrimitive(runtime),
+      realBias.getPrimitive(runtime))
+    val indexes = Array.fill(srcs.length)(0)
+
+    val dsts = Array(realDst.getPrimitive(runtime), realDst_iter.getPrimitive(runtime))
+
+    val primitive = MklDnn.PrimitiveCreate2(fwdPD, srcs, indexes, srcs.length, dsts, dsts.length)
+
+    updateOutputMemoryPrimitives = srcs ++ dsts
+    updateOutputPrimitives = Array(primitive)
+    output = initTensor(realDst)
+
+    _inputFormats = Array(realSrc)
+    _outputFormats = Array(realDst)
+
+    (_inputFormats, _outputFormats)
+  }
+
+  override def updateOutput(input: Activity): Activity = {
+    if (updateOutputTensors == null) {
+      val buffer = new ArrayBuffer[Tensor[Float]]()
+      buffer.append(input.asInstanceOf[Tensor[Float]])
+      buffer.append(src_i.native)
+      buffer.append(weight.native)
+      buffer.append(weight_i.native)
+      buffer.append(bias.native)
+      buffer.append(output.asInstanceOf[Tensor[Float]])
+      buffer.append(dst_i.native)
+
+      updateOutputTensors = buffer.toArray
+    }
+
+    updateWithNewTensor(updateOutputTensors, 0, input)
+
+    MklDnnOps.streamSubmit(runtime.stream, 1, updateOutputPrimitives, updateOutputPrimitives.length,
+      updateOutputMemoryPrimitives, updateOutputTensors)
+
+    output
+  }
+
+  override private[mkldnn] def initBwdPrimitives(grad: Array[MemoryData], phase: Phase) = {
+    _gradInputFormats = grad.clone()
```

Comment:
If this layer does not support backward, you should throw exception if calling backward.